parastell/source_mesh.py

```diff
@@ -375,23 +413,38 @@ def _create_tets_from_hex(self, s_idx, theta_idx, phi_idx):
         # first, ordered clockwise relative to the thumb, followed by the
         # remaining vertex at the end of the thumb.
         # See Moreno, Bader, Wilson 2024 for hexahedron splitting
-        hex_canon_ids = [
-            [idx_list[0], idx_list[2], idx_list[1], idx_list[5]],
-            [idx_list[0], idx_list[3], idx_list[2], idx_list[7]],
-            [idx_list[0], idx_list[7], idx_list[5], idx_list[4]],
-            [idx_list[7], idx_list[2], idx_list[5], idx_list[6]],
-            [idx_list[0], idx_list[2], idx_list[5], idx_list[7]],
+        canonical_ordering_schemes = [
+            [
+                [idx_list[0], idx_list[3], idx_list[1], idx_list[4]],
+                [idx_list[1], idx_list[3], idx_list[2], idx_list[6]],
+                [idx_list[1], idx_list[4], idx_list[6], idx_list[5]],
+                [idx_list[3], idx_list[6], idx_list[4], idx_list[7]],
+                [idx_list[1], idx_list[3], idx_list[6], idx_list[4]],
+            ],
+            [
+                [idx_list[0], idx_list[2], idx_list[1], idx_list[5]],
+                [idx_list[0], idx_list[3], idx_list[2], idx_list[7]],
+                [idx_list[0], idx_list[7], idx_list[5], idx_list[4]],
+                [idx_list[7], idx_list[2], idx_list[5], idx_list[6]],
+                [idx_list[0], idx_list[2], idx_list[5], idx_list[7]],
+            ],
         ]
 
-        for vertex_ids in hex_canon_ids:
+        # Alternate canonical ordering schemes defining hexahedron splitting to
+        # avoid gaps and overlaps between non-planar hexahedron faces
+        # s_idx begins at 0 and theta_idx begins at 1
+        scheme_idx = ((s_idx + 1) + (theta_idx - 1) + phi_idx) % 2
+
+        for vertex_ids in canonical_ordering_schemes[scheme_idx]:
             self._create_tet(vertex_ids)
 
     def _create_tets_from_wedge(self, theta_idx, phi_idx):
         """Creates three tetrahedra from defined wedge.
         (Internal function not intended to be called externally)
 
         Arguments:
-            idx_list (list of int): list of wedge vertex indices.
+            theta_idx (int): index defining location along poloidal angle axis.
```

Comment:
I don't understand the stencil below.  Maybe it's been this way for a while, but I don't see any points in this stencil at a `s_idx=1` which I expect.  
I'm also not sure why this goes from [1,num_phi] istead of [0,num_phi-1]

Reply:
Yes, I agree that it's slightly confusing, and our indexing scheme could use some changes to make it more intuitive. The way we've defined things is that the vertex at the magnetic axis and those on the first closed flux surface all have a `s_idx` of 0. This is also why `theta_idx` begins at 1, to differentiate the first vertex on the first CFS from that on the magnetic axis.

It would make much more sense to have `s_idx = 0` to always refer to the magnetic axis vertices, and `s_idx = 1` refer to the first CFS. As such, we could then have `theta_idx` start from 0. I think I'll also change the naming convention from using to `theta_idx` and `phi_idx` to using `poloidal_idx` and `toroidal_idx`, respectively, to avoid confusion.